packages/nimble-components/src/table-column/icon/models/table-column-icon-validator.ts

```diff
@@ -0,0 +1,43 @@
+import type { Mapping } from '../../../mapping/base';
+import { MappingIcon } from '../../../mapping/icon';
+import type { ColumnInternals } from '../../base/models/column-internals';
+import {
+    TableColumnEnumBaseValidator,
+    enumBaseValidityFlagNames
+} from '../../enum-base/models/table-column-enum-base-validator';
+import type { MappingKeyType } from '../../enum-base/types';
+
+const iconValidityFlagNames = [
+    ...enumBaseValidityFlagNames,
+    'invalidIconName'
+] as const;
+
+/**
+ * Validator for TableColumnIcon
+ */
+export class TableColumnIconValidator extends TableColumnEnumBaseValidator<
+    typeof iconValidityFlagNames
+> {
+    public constructor(
+        columnInternals: ColumnInternals<unknown>,
+        supportedMappingElements: readonly (typeof Mapping)[]
+    ) {
+        super(columnInternals, iconValidityFlagNames, supportedMappingElements);
```

Comment:
Do we need to take supportedMappingElements as a constructor parameter? It doesn't vary per instance of TableColumnIconValidator right?

Seems like the icon validator and the text validator can pass the supported mappings array directly to the base, don't need it parameterized by the constructor.